<a href="https://colab.research.google.com/github/tschelli/Project-2-ML-/blob/main/Project_2_Part_1_Proposals.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2 - Proposal Datasets

Coding Dojo Data Science Project 2
- Name: Tyler Schelling
- Date: 10/27/2022

##Mount Drive | Import Libraries | Download Data

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# Imports
import pandas as pd
import numpy as np

##First Choice: Dataset 1 - Stroke Prediction Dataset

In [4]:
stroke_filename = '/content/drive/MyDrive/02. Life/Coding Dojo/00 Datasets/healthcare-dataset-stroke-data.csv'
stroke_df = pd.read_csv(stroke_filename)
stroke_df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [5]:
stroke_df.shape

(5110, 12)

###Dataset Source

The dataset can be found [here](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset).

###Data Description

This dataset is used to predict whether a patient is likely to have a stroke based on the input parameters like gender, age, various diseases, and smoking status. 

Column Name | Description
---|---
id | Unique identifier
gender| 'Male', 'Female', or 'Other'
age| Age of the patient
hypertension | 0 if patient doesn't have hypertension, 1 if the patient has hypertension
ever_married | 'No' or 'Yes'
work_type | 'children', 'govt_job', 'never_worked', 'private', or 'self-employed'
Residence_type | 'Rural' or 'Urban'
avg_glucose_level | average glucose level in blood
bmi | body mass index
smoking_status | 'formerly smoke', 'never smoked', 'smokes', or 'unknown'
stroke | 1 if the patient had a stroke or 0 if not

###Target

The target will be the `stroke` column to predict whether or not the patient had a stroke.

###Row Represents

A row represents a patient who has or has not had a stroke.

###Type of Problem

This will be a classification problem to determine if a patient had a stroke or not.

###Number of Features

There are 11 features associated with the dataset. However, some features may not be applicable to machine learning modeling.

###Number of Rows

There are 5110 rows in the dataset.

###Potential Challenges

In [16]:
stroke_df['stroke'].value_counts()

0    4861
1     249
Name: stroke, dtype: int64

- Unknown smoking status may have a negative impact on the predictive capabilities of the model.
- Largely unbalanced target variable in the dataset.

##Second Choice: Dataset 2 - Cirrhosis Prediction Dataset

In [12]:
cirr_filename = '/content/drive/MyDrive/02. Life/Coding Dojo/00 Datasets/cirrhosis.csv'
cirr_df = pd.read_csv(car_filename)
cirr_df.head()

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

In [13]:
cirr_df.shape

(418, 20)

###Dataset Source

The dataset can be found [here](https://www.kaggle.com/datasets/fedesoriano/cirrhosis-prediction-dataset).

###Data Description

This dataset is used to predict which stage of cirrhosis a patient has based on various attributes and data collected by each patient.

Column name | Description 
---|---
ID| unique identifier
N-Days | # of days between registration and the earlier of death, transplantation, or study analysis time in July 1986
Status | status of the patient C (censored), CL (censored due to liver tx), or D (death)
Drug| type of drug D-penicillamine or placebo
Age|age in days
Sex|M(male) or F (female)
Acites| presence of ascites N (no) or Y (yes)
Hepatomegaly | presence of hepatomegaly N (no) or Y (yes)
Spiders | presence of spiders N (no) or Y (yes)
Edema | presence of edema N (no edema and no diuretic therapy for edema), S (edema present without diuretics, or edema resolved by diuretics), or Y (edema despite diuretic therapy)
Bilirubin| serum bilirubin in [mg/dl]
Cholesterol | serum cholesterol in [mg/dl]
Albumin | albumin in [gm/dl]
Copper | urine copper in [ug/day]
Alk_Phos | alkaline phosphatase in [U/liter]
SGOT| SGOT in [U/ml]
Triglycerides| triglicerides in [mg/dl]
Platelets | platelets per cubic [ml/1000]
Prothrombin| prothrombin time in seconds 
Stage | histologic stage of disease (1, 2, 3, or 4)

###Target

The target variable will be the `stage` column to predict what stage of the disease a patient has.

###Row Represents

A single row represents a patient with Cirrhosis.

###Type of Problem

This will be a classification problem to determine what stage of cirrhosis the patient has.

###Number of Features

There are 19 features associated with the dataset. However, some features may not be applicable to machine learning modeling.

###Number of Rows

There are 418 rows in the dataset.

###Potential Challenges

In [15]:
cirr_df['Stage'].value_counts()

3.0    155
4.0    144
2.0     92
1.0     21
Name: Stage, dtype: int64

- Limited amount of data.
- Balance across the stage types (target) is skewed.